### PR TITLE
Improved server side JSON validation

### PR DESCRIPTION
**Pull request by mbarton**
Refactored out JSON validation in `Api2` into a single method, which returns a `BadRequest` response detailing the errors.

For example, missing out fields in the `create` UI no longer causes a 500 but returns a list of the missing fields:

```
/posterImage -> ValidationError(List(error.path.missing),WrappedArray())
/youtubeCategoryId -> ValidationError(List(error.path.missing),WrappedArray())
/channelId -> ValidationError(List(error.path.missing),WrappedArray())
```

Also update the front-end to clear out saving/publishing/searching state when an error occurs. This stops the save button from spinning forever.

NB: the error still shows up in red at the top forever (#187)

### CODE REVIEW

File: app/controllers/Api2.scala [hunk 1]
```diff
@@ -70,69 +70,47 @@ class Api2 @Inject() (implicit val previewDataStore: PreviewDataStore,
...
     implicit val user = req.user
-    try {
-      val updatedAtom = PublishAtomCommand(id).process()
-      Ok(Json.toJson(updatedAtom))
-    } catch {
-      commandExceptionAsResult
-    }
+
+    val updatedAtom = PublishAtomCommand(id).process()
+    Ok(Json.toJson(updatedAtom))
```
Comment (akash1810) [inline app/controllers/Api2.scala]: as we've removed the `try/catch` what happens when `PublishAtomCommand(id).process()` do we get a 500 and a sensible message still?

File: app/controllers/Api2.scala [hunk 2]
```diff
@@ -141,35 +119,47 @@ class Api2 @Inject() (implicit val previewDataStore: PreviewDataStore,
...
+      val atom = command.process()
+      Ok(Json.toJson(atom))
     }
   }
 
   def getAuditTrailForAtomId(id: String) = APIHMACAuthAction { implicit req =>
     Ok(Json.toJson(auditDataStore.getAuditTrailForAtomId(id)))
   }
+
+  private def parse[T](raw: UserRequest[AnyContent])(fn: T => Result)(implicit reads: Reads[T]): Result = try {
```
Comment (akash1810) [inline app/controllers/Api2.scala]: nice

File: app/controllers/Api2.scala (hunk 1 above)
Comment (mbarton) [inline app/controllers/Api2.scala]: Oops sorry. I've put back the try/catch

File: app/controllers/Api2.scala [hunk 3]
```diff
@@ -2,23 +2,23 @@ package controllers
 
 import javax.inject.Inject
 
+import _root_.util.{AWSConfig, ExpiryPoller, YouTubeConfig, YouTubeVideoUpdateApi}
```
Comment (akash1810) [inline app/controllers/Api2.scala]: `_root_` silly intellij!

Comment (mbarton) [inline app/controllers/Api2.scala]: Yeah. I double checked it was like that before I changed things

**Pull request merged**